automation/jenkins/aws/setCredentials.sh

```diff
@@ -6,78 +6,152 @@
+                            --role-session-name "$(echo $GIT_USER | tr -cd '[[:alnum:]]' )" \
+                            --output json > ${TEMP_CREDENTIAL_FILE}
+                    unset AWS_ACCESS_KEY_ID
+                    unset AWS_SECRET_ACCESS_KEY
+                    AWS_CRED_TEMP_AWS_ACCESS_KEY_ID=$(jq -r '.Credentials.AccessKeyId' < ${TEMP_CREDENTIAL_FILE})
+                    AWS_CRED_TEMP_AWS_SECRET_ACCESS_KEY=$(jq -r '.Credentials.SecretAccessKey' < ${TEMP_CREDENTIAL_FILE})
+                    AWS_CRED_TEMP_AWS_SESSION_TOKEN=$(jq -r '.Credentials.SessionToken' < ${TEMP_CREDENTIAL_FILE})
+                    rm ${TEMP_CREDENTIAL_FILE}
+                fi
             fi
         fi
+        ;;
 
-        if [[ ("${!AWS_CRED_AUTOMATION_USER_VAR^^}" == "ROLE") ||
-              ((-n ${AWS_CRED_AWS_ACCESS_KEY_ID}) && (-n ${AWS_CRED_AWS_SECRET_ACCESS_KEY})) ]]; then
-            TEMP_CREDENTIAL_FILE="${AUTOMATION_DATA_DIR}/temp_aws_credentials.json"
-            unset AWS_SESSION_TOKEN
-
-            aws sts assume-role \
-                --role-arn arn:aws:iam::${!AWS_CRED_AWS_ACCOUNT_ID_VAR}:role/${AWS_CRED_AUTOMATION_ROLE} \
-                --role-session-name "$(echo $GIT_USER | tr -cd '[[:alnum:]]' )" \
-                --duration-seconds "${AUTOMATION_ROLE_VALIDITY}" \
-                --output json > ${TEMP_CREDENTIAL_FILE} ||
-                aws sts assume-role \
-                    --role-arn arn:aws:iam::${!AWS_CRED_AWS_ACCOUNT_ID_VAR}:role/${AWS_CRED_AUTOMATION_ROLE} \
-                    --role-session-name "$(echo $GIT_USER | tr -cd '[[:alnum:]]' )" \
-                    --output json > ${TEMP_CREDENTIAL_FILE}
-            unset AWS_ACCESS_KEY_ID
-            unset AWS_SECRET_ACCESS_KEY
-            AWS_CRED_TEMP_AWS_ACCESS_KEY_ID=$(jq -r '.Credentials.AccessKeyId' < ${TEMP_CREDENTIAL_FILE})
-            AWS_CRED_TEMP_AWS_SECRET_ACCESS_KEY=$(jq -r '.Credentials.SecretAccessKey' < ${TEMP_CREDENTIAL_FILE})
-            AWS_CRED_TEMP_AWS_SESSION_TOKEN=$(jq -r '.Credentials.SessionToken' < ${TEMP_CREDENTIAL_FILE})
-            rm ${TEMP_CREDENTIAL_FILE}
+    azure)
+
+        # find the method - default to prompt based login
+        AZ_AUTOMATION_AUTH_METHOD_VAR="${CRED_ACCOUNT}_AZ_AUTH_METHOD"
+        if [[ -z "${!AZ_AUTOMATION_AUTH_METHOD_VAR}" ]]; then AZ_AUTOMATION_AUTH_METHOD_VAR="AZ_AUTH_METHOD"; fi
+        AZ_AUTH_METHOD="${!AZ_AUTOMATION_AUTH_METHOD_VAR:-interactive}"
+
+        # lookup the AZ Account
+        AZ_ACCOUNT_ID_VAR="${CRED_ACCOUNT}_AZ_ACCOUNT_ID"
+        if [[ -n "${!AZ_ACCOUNT_ID_VAR}" ]]; then
+            AZ_ACCOUNT_ID="${AZ_ACCOUNT_ID_VAR}"
         fi
-    fi
-fi
+
+        # Set the tenant - this can be account specific but most likely global
+        AZ_ACCOUNT_TENANT_OVERRIDE_VAR="${CRED_ACCOUNT}_AZ_TENANT_ID"
+        if [[ -n "${!AZ_ACCOUNT_TENANT_OVERRIDE_VAR}" ]]; then
+            AZ_TENANT_ID="${AZ_ACCOUNT_TENANT_OVERRIDE_VAR}"
```

Comment:
He as well. It can only be set via this if-statement so in the event the condition isn't met, it'll fall over during the CLI call and be harder to diagnose.

Reply:
This one is actually optional so will make it an additional arg on the cli. Also neds to be added to all logins

Reply:
Looks good